### PR TITLE
DOC: fix Subplot calls

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

There were a few calls to `plt.Subplot` that confused the doc builds (and me!  See long discussion https://github.com/sphinx-gallery/sphinx-gallery/pull/448).  I don't even understand how those calls worked, since `pyplot` does not have anything defined as `Subplot`.  But basically they clobbered the `pyplot.subplot` gallery tracking.  Changed these instances to `add_subplot` and all seems to build fine.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by anntzer** [comment]
anyone can merge post ci

**Comment by jklymak** [comment]
@anntzer do you understand where Subplot comes from?

**Comment by larsoner** [comment]
FYI you should also be able to remove the `numpy ` and `scipy` entries from `reference_url` entry from your `sphinx_gallery_conf` since nowadays SG uses `intersphinx`.

**Comment by anntzer** [comment]
That's the subclass of Axes actually returned by add_subplot(), but in practice no one ever calls it directly.

File: examples/subplots_axes_and_figures/gridspec_nested.py [hunk 1]
```diff
@@ -24,22 +24,16 @@ def format_axes(fig):
...
-f.add_subplot(ax5)
-ax6 = plt.Subplot(f, gs01[-1, -1])
-f.add_subplot(ax6)
+ax1 = f.add_subplot(gs00[:-1, :])
+ax2 = f.add_subplot(gs00[-1, :-1])
+ax3 = f.add_subplot(gs00[-1, -1])
+
+# the following syntax is the same as above, except perhaps more convenient:
+gs01 = gs0[1].subgridspec(3, 3)
+
```
Comment (QuLogic) [inline examples/subplots_axes_and_figures/gridspec_nested.py]: Can you remove this newline and the one after `gs00` above? Or add a comment above `gs00`?
When I first read the comment about "same as above", I thought it was about `add_subplot`, but it's really talking about the `gridspec.GridSpecFromSubplotSpec(...)`.

Comment (jklymak) [reply examples/subplots_axes_and_figures/gridspec_nested.py]: Rephrased to `# the following syntax does the same as the GridSpecFromSubplotSpec call above:`

**Pull request merged**